Extends Org Create endpont + shared secret auth

ikreymer opened the pull request:
Updates the /api/orgs/create endpoint to:
- not have name / slug be required, will be renamed on first user via #1870 
- support optional quotas
- support optional first admin user email, who will receive an invite to join the org.

Also supports a new shared secret mechanism, to allow an external automation to access the /api/orgs/create endpoint (and only that endpoint thus far) via a shared secret instead of normal login.

ikreymer committed:
extend /api/orgs/create endpoint to also accept first user to invite, quotas,
and make name optional to support automated usage

ikreymer committed:
allow access to certain apis as superadmin
as superadmin if a known shared secret is passed as auth token
only enabled for /api/orgs/create for now to allow programmatic
access to this api

ikreymer committed:
lint fix

ikreymer committed:
fix typo

tw4l commented on backend/btrixcloud/orgs.py:
```diff
@@ -801,27 +802,53 @@ async def get_orgs(
+            ):
+                result["invited"] = "new_user"
+            else:
+                result["invited"] = "existing_user"
```
In this case are we actually inviting an existing user yet or just creating the org without sending any invitations yet at this stage? I think the latter, yeah? In that case it might make sense to just omit this or set the value to `None`?

ikreymer replied:
yes, `invited` is set if `new_user` or `existing_user` is invited, otherwise it will be unset (undefined in response).

ikreymer replied:
this is just reusing the return values from the invite endpoint, with possibility of it being unset as well.

tw4l reviewed (commented):
Looks great so far, just one relatively minor comment.

We should add some tests with the various values present and missing and test that the org is created and configured the way we expect.

ikreymer commented:
> Looks great so far, just one relatively minor comment.
> 
> We should add some tests with the various values present and missing and test that the org is created and configured the way we expect.

yeah, probably should add a few more tests, including for automatic renaming when first admin user joins.

ikreymer committed:
use BTRIX_SUBS_APP_API_KEY and load from optional 'btrix-subs-app-secret',
to be provided externally

ikreymer committed:
fix default port

ikreymer committed:
fix template

ikreymer committed:
tests: add test for creating new org with:
- inviting new user
- inviting existing user
- quotas
- automatic org renaming if slug/name not set
fixes: return invite token id (no way to get it again for existing user), return OrgOut instead of Org

ikreymer committed:
org accept-invite path should not accept an invite with no org

tw4l commented on backend/test/test_org.py:
```diff
@@ -555,3 +559,147 @@ def test_update_read_only(admin_auth_headers, default_org_id):
+    data = r.json()
+    assert data["added"]
+
+    org_id = data["id"]
```
Maybe worth testing that the new org's name is a string representation of its id at this point?

ikreymer replied:
Will update in follow-up PR, which adds additional tests.

tw4l reviewed (approved):
Very nice! Thanks for adding tests. Not sure how you want to handle https://github.com/webrecorder/browsertrix/pull/1902, but I think merging this first and then making changes in that follow-up PR seems reasonable?
